peers/app_request_network.go

```diff
@@ -191,12 +191,15 @@ func (n *AppRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) set.Set[id
 	// In this case, we need to manually track the API node.
 	startInfoAPICall = time.Now()
 	if apiNodeID, _, err := n.infoAPI.GetNodeID(context.Background()); err != nil {
+		n.setInfoAPICallLatencyMS(float64(time.Since(startInfoAPICall).Milliseconds()))
+
 		n.logger.Error(
 			"Failed to get API Node ID",
 			zap.Error(err),
 		)
-		n.setInfoAPICallLatencyMS(float64(time.Since(startInfoAPICall).Milliseconds()))
 	} else if nodeIDs.Contains(apiNodeID) {
+		n.setInfoAPICallLatencyMS(float64(time.Since(startInfoAPICall).Milliseconds()))
+
```

Comment:
This control flow is a bit hard to follow. I'd recomment moving the call to `infoAPI.GetNodeID` above the `if` statement to clean it up:
```
startInfoAPICall = time.Now()
apiNodeID, _, err := n.infoAPI.GetNodeID(context.Background())
n.setInfoAPICallLatencyMS(float64(time.Since(startInfoAPICall).Milliseconds()))
if err != nil {
  ...
} else if nodeIDs.Contains(apiNodeID) {
  ...
} 
```

Similarly for the below call to `infoAPI.GetNodeIP`, let's move it out of the `if` statement so that we can put the latency measurement right next to the call:
```
startInfoAPICall = time.Now()
apiNodeIP, err := n.infoAPI.GetNodeIP(context.Background())
n.setInfoAPICallLatencyMS(float64(time.Since(startInfoAPICall).Milliseconds()))
if err != nil {
 ...
}
```